### PR TITLE
Add testing instance to shinyapps.io deploy

### PR DESCRIPTION
**Pull request by afwillia**
This github workflow file for the shinyapps.io deploy adds a new testing instance. There are now 3 deployment types:

1. Release - triggered by a tagged commit. Deploys the tagged DCA branch and installs schematic from pypi
2. Staging - triggered by an untagged commit to main. Deploys the main DCA branch and installs schematic from pypi
3. Testing - triggered by an untagged commit to develop* branches. Deploys the branch and installs schematic develop branch using poetry.

### CODE REVIEW

**Comment by rrchai** [comment]
@afwillia That's great work. I only have two minor suggestions. Otherwise, this PR looks good to me. 

Remember to update the branch before merging, and use `squash merge` for feature branch to get cleaner commit history.

**Review by rrchai — APPROVED** [review]

**Comment by rrchai** [comment]
@afwillia That's great work. I only have two minor suggestions. Otherwise, this PR looks good to me.

Remember to update the branch before merging, and use squash merge for feature branch to get cleaner commit history.

**Comment by rrchai** [comment]
Linked to issues:
- fixes #363
- fixes #365

File: .github/workflows/shinyapps_deploy.yml [hunk 1]
```diff
@@ -90,12 +98,15 @@ jobs:
         run: |
           # if there is a tag, 'refName' will be tag name
           refName <- Sys.getenv("GITHUB_REF_NAME")
+          # if tag is v*.*.*, deploy to prod, if main to staging, otherwise to test
+          if (!grepl("v[0-9]+.[0-9]+.[0-9]+", refName)) {
+            release_type <- "release"
```
[the remark this answers is not in the document]
Comment (rrchai) [reply .github/workflows/shinyapps_deploy.yml:103]: move the comment to #363

File: .github/workflows/shinyapps_deploy.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ on:
   push:
     branches:
       - main
-      - develop
+      - develop*
```
[the remark this answers is not in the document]
Comment (rrchai) [reply .github/workflows/shinyapps_deploy.yml:12]: move the comment to #363

**Comment by afwillia** [comment]
Merging the new workflow into develop after addressing minor comments from @rrchai

**Pull request merged**